src/accounts/accounts.controller.ts

```diff
@@ -0,0 +1,145 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  Param,
+  Delete,
+  HttpException,
+  ParseUUIDPipe,
+} from '@nestjs/common';
+import { DepositTransactionDto } from 'src/transactions/dto/deposit-transaction.dto';
+import { WithdrawTransactionDto } from 'src/transactions/dto/withdraw-transaction.dto';
+import { SwyftTablesInfo } from 'src/typings/types';
+import { CreateTransactionDto } from '../transactions/dto/create-transaction.dto';
+import { Transaction } from '../transactions/entities/transaction.entity';
+import { AccountsService } from './accounts.service';
+import { CreateAccountDto } from './dto/create-account.dto';
+import { UpdateAccountDto } from './dto/update-account.dto';
+import { Account } from './entities/account.entity';
+import { ValidationPipe } from './pipes/swyft-validation.pipe';
+
+@Controller('accounts')
+export class AccountsController {
+  constructor(private readonly accountsService: AccountsService) {}
+
+  /**
+   * Handler for creating an account
+   * @param createAccountDto payload
+   * @returns
+   */
+  @Post()
+  async createAccount(
+    @Body(new ValidationPipe()) createAccountDto: CreateAccountDto
+  ): Promise<HttpException> {
+    return this.accountsService.createAccount(createAccountDto);
+  }
+
+  /**
+   * Handler for retrieving all accounts information
+   * @returns
+   */
+  @Get()
+  async findAllAccounts(): Promise<SwyftTablesInfo | HttpException> {
+    return this.accountsService.findAllAccounts();
+  }
+
+  /**
+   * Handler for retrieving individual account information
+   * @param accountId Account ID of the target account for retrieval
+   * @returns
+   */
+  @Get(':id')
+  async findOneAccount(
+    @Param('id') accountId: string
+  ): Promise<Array<Account | Transaction> | HttpException> {
+    return this.accountsService.findOneAccount(accountId);
+  }
+
+  /**
+   * Handler for account update
+   * @param accountId Account ID of the target account for update
+   * @param updateAccountDto payload
+   * @returns
+   */
+  @Patch(':id')
+  async updateAccount(
+    @Param('id') accountId: string,
```

Comment:
@Param('id', new ParseUUIDPipe()) accountId: string,

add this whenever you have a uuid param

Reply:
Awesome. Is there a difference between calling ParseUUIDPipe and new ParseUUIDPipe(). The ParseUUIDPipe works for me.